include/linux/kfuzztest.h

```diff
@@ -0,0 +1,449 @@
+		test_arg_type *arg, struct reloc_region_array *regions);       \
+	const struct kfuzztest_target __fuzz_test__##test_name __attribute__(( \
+		__section__(".kfuzztest_target"), __used__)) = {               \
+		.name = #test_name,                                            \
+		.arg_type_name = #test_arg_type,                               \
+		.write_input_cb = kfuzztest_write_cb_##test_name,              \
+	};                                                                     \
+	static ssize_t kfuzztest_write_cb_##test_name(struct file *filp,       \
+						      const char __user *buf,  \
+						      size_t len, loff_t *off) \
+	{                                                                      \
+		int ret;                                                       \
+		struct reloc_region_array *regions;                            \
+		struct reloc_table *rt;                                        \
+		void *payload_start, *payload_end, *buffer;                    \
+		test_arg_type *arg;                                            \
+                                                                               \
+		buffer = kmalloc(len, GFP_KERNEL);                             \
+		if (!buffer)                                                   \
+			return -ENOMEM;                                        \
+		else if (IS_ERR(buffer))                                       \
+			return PTR_ERR(buffer);                                \
+		ret = __kfuzztest_write_cb_common(filp, buf, len, off, buffer, \
+						  len);                        \
+		if (ret)                                                       \
+			goto out;                                              \
+		ret = __kfuzztest_parse_input(buffer, len, &regions, &rt,      \
+					      &payload_start, &payload_end);   \
+		if (ret)                                                       \
+			goto out;                                              \
+		ret = __kfuzztest_relocate(regions, rt, payload_start,         \
+					   payload_end);                       \
+		if (ret)                                                       \
+			goto out;                                              \
+		/* Call the fuzz logic on the provided written input. */       \
+		arg = (test_arg_type *)payload_start;                          \
+		kfuzztest_logic_##test_name(arg, regions);                     \
+		ret = len;                                                     \
+out:                                                                           \
+		kfree(buffer);                                                 \
+		return ret;                                                    \
+	}                                                                      \
+	static void kfuzztest_logic_##test_name(                               \
+		test_arg_type *arg, struct reloc_region_array *regions)
+
+enum kfuzztest_constraint_type : uint8_t {
+	EXPECT_EQ = 0,
+	EXPECT_NE,
```

Comment:
For the sake of complexity, maybe add GE and LT as well? That's not too much code.